### PR TITLE
fix: Disable common-instancetypes feature gate before tests

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does
Before this PR:

The common-instancetypes feature gate is disabled by default but could be enabled during the deployment of KubeVirt.

After this PR:

Disable the common-instancetypes feature gate explicitly before running virt-operator common-instancetypes deployment tests to make sure the feature gate was not enabled before (e.g. in downstream deployments).

Also, improve the definition of vars/consts a little by using blocks.

Fixes:
https://issues.redhat.com/browse/CNV-35336
https://issues.redhat.com/browse/CNV-35337
https://issues.redhat.com/browse/CNV-35333

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by fabiand** [comment]
When do we plan to remove the common instanceTypes FG?

**Comment by lyarwood** [comment]
> When do we plan to remove the common instanceTypes FG?

I was waiting to see the outcome of https://github.com/kubevirt/community/pull/251 before committing to this but I wanted to move the [FG](https://github.com/kubevirt/kubevirt/blob/8f6d1505e7211f0a46ae935ac0457954cb61f761/pkg/virt-config/feature-gates.go#L74-L78) to beta in 1.3 and GA (removal and on by default) by 1.4.

/lgtm
/cherry-pick release-1.2

@0xFelix can you add a link to one or all of the public downstream bugs for this?

**Comment by lyarwood** [comment]
> > When do we plan to remove the common instanceTypes FG?
> 
> I was waiting to see the outcome of [kubevirt/community#251](https://github.com/kubevirt/community/pull/251) before committing to this but I wanted to move the [FG](https://github.com/kubevirt/kubevirt/blob/8f6d1505e7211f0a46ae935ac0457954cb61f761/pkg/virt-config/feature-gates.go#L74-L78) to beta in 1.3 and GA (FG deprecated for removal and on by default) by 1.4 with full FG removal in 1.5.

I've pushed https://github.com/kubevirt/kubevirt/pull/11681 to see if we can move this retrospectively to beta in 1.2 so we can GA and remove the FG by 1.4.

**Comment by 0xFelix** [comment]
> @0xFelix can you add a link to one or all of the public downstream bugs for this?

@lyarwood Done

**Comment by alicefr** [comment]
@0xFelix the only a super nit. I would split the changes in 2 commits. One the const and one with the feature gate enablement.
/approve
/hold
Put on hold if you want to fix the commits, otherwise, please unhold

File: tests/operator/operator.go [hunk 1]
```diff
@@ -2984,6 +2991,7 @@ spec:
 		})
 
 		AfterEach(func() {
+			// Reset the feature gate to its default value after running any of these tests.
```
Comment (alicefr) [inline tests/operator/operator.go:2994]: One question shouldn't this restore the feature gates. What if the feature gate was enabled?

**Comment by alicefr** [comment]
@0xFelix also a question about if the `AfterEach` is correct

File: tests/operator/operator.go (hunk 1 above)
Comment (0xFelix) [reply tests/operator/operator.go]: You are right! Updated the PR.

File: tests/operator/operator.go [hunk 2]
```diff
@@ -2972,10 +2972,20 @@ spec:
...
-		var appComponent string
-		var labelSelector string
+		var (
+			fgEnabled     bool
+			appComponent  string
+			labelSelector string
+		)
 
 		BeforeEach(func() {
+			if fgEnabled = checks.HasFeature(virtconfig.PersistentReservation); fgEnabled {
```
Comment (lyarwood) [inline tests/operator/operator.go:2982]: whops! This needs to be changed.

**Pull request merged**